dotnet/src/Agents/Core/ChatCompletionAgent.cs

```diff
@@ -34,14 +34,25 @@ public sealed class ChatCompletionAgent : ChatHistoryKernelAgent
         }
         chat.AddRange(history);
 
-        var messages =
+        int messageCount = chat.Count;
+
+        IReadOnlyList<ChatMessageContent> messages =
             await chatCompletionService.GetChatMessageContentsAsync(
                 chat,
                 this.ExecutionSettings,
                 this.Kernel,
                 cancellationToken).ConfigureAwait(false);
 
-        foreach (var message in messages ?? [])
+        for (int messageIndex = messageCount; messageIndex < chat.Count; messageIndex++)
```

Comment:
nit: messageIndex < messageCount

Reply:
Updated

Reply:
That's a wrong advise. Please rollback it if it's applied.

Reply:
Oh right...good catch.

Reply:
Rolled-back.